mitodl/mit-open
# Email for saved searches

### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/5578

### Description (What does it do?)
This PR tweaks the emails sent out for search subscription emails and also activates the search subscription email as a celery task.

### Screenshots (if appropriate):
<img width="823" alt="Screenshot 2024-09-27 at 4 33 07 PM" src="https://github.com/user-attachments/assets/ac167cb5-a99b-4f5b-b4c0-8a8d2d1c85f0">
<img width="814" alt="Screenshot 2024-09-27 at 4 33 15 PM" src="https://github.com/user-attachments/assets/ee57f66c-4990-4714-897a-0f47629c758f">



### How can this be tested?
1. checkout this branch.
3. go to RC env var settings in heroku and copy down the following into your local settings:
MAILGUN_SENDER_DOMAIN
MAILGUN_KEY
MAILGUN_FROM_EMAIL
NOTIFICATION_EMAIL_BACKEND
4. restart celery via ```docker compose restart celery```
5. created a saved search subscription that catches learning resources from any provider via the following django shell script (replace user with your own email)
```
from django.contrib.auth import get_user_model
from learning_resources_search.models import PercolateQuery
from learning_resources_search.api import subscribe_user_to_search_query

User = get_user_model()
migrated_user = User.objects.get(email='ambady@mit.edu')
subscribe_user_to_search_query(migrated_user, {"offered_by": ["mitpe", "mitx", "ocw", "see"]}, PercolateQuery.SEARCH_SUBSCRIPTION_TYPE)
```
6. Change the "created_on" date locally for a few learning resources. I usually manually do it via this django script: 
```
import datetime
delta = datetime.timedelta(days=1)
since = now_in_utc() - delta


resources = LearningResource.objects.filter(published=True)[50:53]

for resource in resources:
	resource.created_on = since  +  datetime.timedelta(hours=1)
	resource.save()
```
6. manually send out subscription emails for search subscriptions (make sure the email you have locally is an inbox you have access to):
```
from learning_resources_search.models import PercolateQuery
from learning_resources_search.tasks import send_subscription_emails
send_subscription_emails(PercolateQuery.SEARCH_SUBSCRIPTION_TYPE)
```
